[15.0][MIG] hr_timesheet_activity_begin_end

CRogos opened the pull request:
contains changes of #507 and depends on hr_timesheet_sheet

Commit:
Rename module to hr_timesheet_activity_begin_end

luismontalba committed:
[MIG] hr_timesheet_activity_begin_end: Migrated to 10.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: hr-timesheet-10.0/hr-timesheet-10.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/hr-timesheet-10-0/hr-timesheet-10-0-hr_timesheet_activity_begin_end/it/

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: hr-timesheet-10.0/hr-timesheet-10.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/hr-timesheet-10-0/hr-timesheet-10-0-hr_timesheet_activity_begin_end/es/

ernestotejeda committed:
[MIG] hr_timesheet_activity_begin_end: Migration to 12.0

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: timesheet-12.0/timesheet-12.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/timesheet-12-0/timesheet-12-0-hr_timesheet_activity_begin_end/

marylla committed:
Added translation using Weblate (German)

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (6 of 6 strings)

Translation: timesheet-12.0/timesheet-12.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/timesheet-12-0/timesheet-12-0-hr_timesheet_activity_begin_end/de/

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: timesheet-12.0/timesheet-12.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/timesheet-12-0/timesheet-12-0-hr_timesheet_activity_begin_end/

astirpe committed:
[FIX] Avoid error on merge timesheet sheet lines

astirpe committed:
[IMP] Sorting timesheets by date and time

OCA-git-bot committed:
hr_timesheet_activity_begin_end 12.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

OCA-git-bot committed:
hr_timesheet_activity_begin_end 12.0.1.1.0

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

davejames committed:
[IMP] hr_timesheet_activity_begin_end: black, isort, prettier

davejames committed:
[MIG] hr_timesheet_activity_begin_end: Migration to 13.0

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: timesheet-13.0/timesheet-13.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/timesheet-13-0/timesheet-13-0-hr_timesheet_activity_begin_end/

davejames committed:
[FIX] hr_timesheet_activity_begin_end: fix false positives in _check_time_start_stop constraint due to float precision

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

OCA-git-bot committed:
hr_timesheet_activity_begin_end 13.0.1.0.1

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: timesheet-13.0/timesheet-13.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/timesheet-13-0/timesheet-13-0-hr_timesheet_activity_begin_end/

schhatbar-initos committed:
[IMP] hr_timesheet_activity_begin_end: black, isort, prettier

schhatbar-initos committed:
[14.0][MIG]hr_timesheet_activity_begin_end

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update hr_timesheet_activity_begin_end.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: timesheet-14.0/timesheet-14.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/timesheet-14-0/timesheet-14-0-hr_timesheet_activity_begin_end/

simonstromb committed:
Added translation using Weblate (Swedish)

simonstromb committed:
Added translation using Weblate (Swedish)

simonstromb committed:
Translated using Weblate (Swedish)

Currently translated at 88.8% (8 of 9 strings)

Translation: timesheet-14.0/timesheet-14.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/timesheet-14-0/timesheet-14-0-hr_timesheet_activity_begin_end/sv/

simonstromb committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (9 of 9 strings)

Translation: timesheet-14.0/timesheet-14.0-hr_timesheet_activity_begin_end
Translate-URL: https://translation.odoo-community.org/projects/timesheet-14-0/timesheet-14-0-hr_timesheet_activity_begin_end/sv/

Commit:
[IMP] hr_timesheet_activity_begin_end: black, isort, prettier

CRogos commented:
@schhatbar-initos @bosd could you have a look at this? You were involved in the v14 migration.

manuelcalerosolis commented:
Trying to download the module gives me this error.

Auto-merging setup/_metapackage/setup.py507)
**CONFLICT (content): Merge conflict in setup/_metapackage/setup.py**
Automatic merge failed; fix conflicts and then commit the result.

setup/_metapackage/setup.py

    install_requires=[
        'odoo-addon-hr_timesheet_task_domain>=15.0dev,<15.1dev',
        'odoo-addon-hr_timesheet_task_required>=15.0dev,<15.1dev',
        'odoo14-addon-hr_timesheet_activity_begin_end',
        'odoo14-addon-hr_timesheet_begin_end',
        'odoo14-addon-hr_timesheet_analysis',
        'odoo14-addon-hr_timesheet_sheet_begin_end',
    ],
    classifiers=[
        'Programming Language :: Python',

CRogos commented:
I have rebased the branch, please try again.

manuelcalerosolis commented:
Now is ok!

Thanks

CRogos commented:
I've rebased and implemented the split/rename of hr_timesheet_activity_begin_end in hr_timesheet_begin_end and hr_timesheet_sheet_begin_end.

```mermaid
  graph TD;
subgraph Active
      A["hr_timesheet"]
      
      A-->B["hr_timesheet_begin_end"];
      A-->C["hr_timesheet_sheet"];
      C-->D["hr_timesheet_sheet_begin_end"];
      B-->D;
end
      subgraph Depricated
          D-->E["hr_timesheet_activity_begin_end"];
      end
```

CRogos commented:
@alexey-pelykh @pedrobaeza could you have a look on this migration. It is open for quite a while and the migration is finished from my perspective.

nimarosa reviewed (approved):
Code review looks okay.

nimarosa commented:
/ocabot merge nobump

nimarosa commented:
/ocabot migration hr_timesheet_activity_begin_end

nimarosa commented:
/ocabot merge nobump

sbidoul commented on setup/hr_timesheet_begin_end/odoo/addons/hr_timesheet_begin_end:
```diff
@@ -0,0 +1 @@
+../../../../hr_timesheet_begin_end
```
This should be a symbolic link, other the merge will fail.

sbidoul commented on setup/hr_timesheet_sheet_begin_end/odoo/addons/hr_timesheet_sheet_begin_end:
```diff
@@ -0,0 +1 @@
+../../../../hr_timesheet_sheet_begin_end
```
This should be a symbolic link, other the merge will fail.

sbidoul reviewed (dismissed):
A technical remark, raised by the bot monitoring.

nimarosa commented:
@CRogos please make the changes suggested by @sbidoul so we can try the merge again

CRogos committed:
[MIG] hr_timesheet_activity_begin_end: Update to 15.0

CRogos commented:
Ok I think I've fixed the links. Is my change causing now the build error? From the logs I assume a github connection issue?

sbidoul reviewed (commented):
All green now, thanks. Please retry the merge

nimarosa commented:
/ocabot merge nobump

EmilioPascual commented:
I have just added rename module to apriory in openupgrade https://github.com/OCA/OpenUpgrade/pull/4340
